File: backend/internal/domain/vehicle.go
package domain

import "time"

type Vehicle struct {
	ID         string    `json:"id"`
	WorkshopID string    `json:"workshop_id"`
	Plate      string    `json:"plate"`
	Brand      string    `json:"brand"`
	Model      string    `json:"model"`
	Year       int       `json:"year"`
	VIN        string    `json:"vin"`
	Color      string    `json:"color"`
	OwnerName  string    `json:"owner_name"`
	OwnerPhone string    `json:"owner_phone"`
	OwnerEmail string    `json:"owner_email"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

File: packages/types/src/vehicle.ts
export interface Vehicle {
  id: string;
  workshop_id: string;
  plate: string;
  brand: string;
  model: string;
  year: number;
  vin: string;
  color: string;
  owner_name: string;
  owner_phone: string;
  owner_email: string;
  created_at: string;
  updated_at: string;
}

export interface CreateVehicleInput {
  plate: string;
  brand?: string;
  model?: string;
  year?: number;
  vin?: string;
  color?: string;
  owner_name?: string;
  owner_phone?: string;
  owner_email?: string;
}
